canvas_modules/common-canvas/src/common-properties/controls/structurelisteditor/structurelisteditor.jsx

```diff
@@ -33,8 +33,7 @@ class StructurelisteditorControl extends AbstractTable {
 
 	addRow() {
 		const newRow = this.getDefaultRow();
-		const rows = this.props.controller.getPropertyValue(this.props.propertyId);
-		rows.push(newRow);
+		const rows = this.props.controller.getPropertyValue(this.props.propertyId).concat([newRow]);
```

Comment:
This is so we get a new property value instead of updating the reference value which causes issues in updatePropertyValue.